src/cla/tcpcl/net.rs

```diff
@@ -0,0 +1,182 @@
+pub(crate) enum TcpClPacket {
+    SessInit(SessInitData),
+    SessTerm(SessTermData),
+    XferSeg(XferSegData),
+    XferAck(XferAckData),
+    XferRefuse(XferRefuseData),
+    KeepAlive,
+    MsgReject(MsgRejectData),
+}
+
+impl TcpClPacket {
+    pub async fn serialize(&self, writer: &mut (impl AsyncWrite + Unpin)) -> anyhow::Result<()> {
+        match self {
+            TcpClPacket::SessInit(sess_init_data) => {
+                writer.write_u8(MessageType::SessInit as u8).await?;
+                writer.write_u16(sess_init_data.keepalive).await?;
+                writer.write_u64(sess_init_data.segment_mru).await?;
+                writer.write_u64(sess_init_data.transfer_mru).await?;
+                writer
+                    .write_u16(sess_init_data.node_id.len() as u16)
+                    .await?;
+                let node_id_bytes = sess_init_data.node_id.as_bytes();
+                writer.write_all(node_id_bytes).await?;
+                writer.write_u32(0).await?;
+            }
+            TcpClPacket::SessTerm(sess_term_data) => {
+                writer.write_u8(MessageType::SessTerm as u8).await?;
+                writer.write_u8(sess_term_data.flags.bits()).await?;
+                writer.write_u8(sess_term_data.reason as u8).await?;
+            }
+            TcpClPacket::XferSeg(xfer_seg_data) => {
+                writer.write_u8(MessageType::XferSegment as u8).await?;
+                writer.write_u8(xfer_seg_data.flags.bits()).await?;
+                writer.write_u64(xfer_seg_data.tid).await?;
+                if xfer_seg_data.flags.contains(XferSegmentFlags::START) {
+                    writer.write_u32(0).await?;
+                }
+                writer.write_u64(xfer_seg_data.len).await?;
+                writer.write_all(xfer_seg_data.buf.as_ref()).await?;
+            }
+            TcpClPacket::XferAck(xfer_ack_data) => {
+                writer.write_u8(MessageType::XferAck as u8).await?;
+                writer.write_u8(xfer_ack_data.flags.bits()).await?;
+                writer.write_u64(xfer_ack_data.tid).await?;
+                writer.write_u64(xfer_ack_data.len).await?;
+            }
+            TcpClPacket::XferRefuse(xfer_refuse_data) => {
+                writer.write_u8(MessageType::XferRefuse as u8).await?;
+                writer.write_u8(xfer_refuse_data.reason as u8).await?;
+                writer.write_u64(xfer_refuse_data.tid).await?;
+            }
+            TcpClPacket::KeepAlive => {
+                writer.write_u8(MessageType::Keepalive as u8).await?;
+            }
+            TcpClPacket::MsgReject(msg_reject_data) => {
+                writer.write_u8(MessageType::MsgReject as u8).await?;
+                writer.write_u8(msg_reject_data.reason as u8).await?;
+                writer.write_u8(msg_reject_data.header).await?;
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn parse(reader: &mut (impl AsyncRead + Unpin)) -> anyhow::Result<Self, TcpClError> {
```

Comment:
as this is the opposite of `serialize` it should be named `deserialize` otherwise you might consider read/write or something similar